File: library/src/main/java/com/getstream/sdk/chat/storage/converter/FilterObjectConverter.java
package com.getstream.sdk.chat.storage.converter;

import androidx.room.TypeConverter;

import com.getstream.sdk.chat.enums.FilterObject;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;

public class FilterObjectConverter {
    static Gson gson = new Gson();

    @TypeConverter
    public static FilterObject stringToObject(String data) {
        if (data == null) {
            return new FilterObject();
        }

        Type listType = new TypeToken<FilterObject>() {
        }.getType();

        return gson.fromJson(data, listType);
    }

    @TypeConverter
    public static String objectToString(FilterObject someObjects) {
        return gson.toJson(someObjects);
    }
}

File: app/src/main/java/com/getstream/sdk/chat/livedata/converter/FilterObjectConverter.kt
package com.getstream.sdk.chat.livedata.converter

import androidx.room.TypeConverter
import com.getstream.sdk.chat.livedata.gson
import com.google.gson.Gson
import io.getstream.chat.android.client.utils.FilterObject
import java.lang.reflect.Type
import com.google.gson.reflect.TypeToken


object FilterObjectConverter {
    @TypeConverter
    @JvmStatic
    fun stringToObject(data: String?): FilterObject {
        if (data == null) {
            return FilterObject()
        }
        val listType: Type = object : TypeToken<FilterObject?>() {}.getType()
        return gson.fromJson(data, listType)
    }

    @TypeConverter
    @JvmStatic
    fun objectToString(someObjects: FilterObject?): String {
        return gson.toJson(someObjects)
    }
}


